Sources/Basic/SignalHandler.swift

```diff
@@ -0,0 +1,110 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import libc
+
+/// The single shared signal manager object, this is initialized here and not in class because we want to reference it from a C function pointer.
+private let sharedSignalManager = SignalManager()
+
+/// The old sigaction to chain the handlers of signals.
+private var oldAction = sigaction()
+
+/// A class to manage POSIX signals.
+public final class SignalManager {
```

Comment:
@ddunbar can you look at this and see if this makes sense, If it does I will create a sep PR for it (with tests) otherwise would appreciate other ideas.